adapters/consumable/consumable.go

```diff
@@ -3,319 +3,124 @@ package consumable
-	}
-
-	if request.Site != nil {
-		body.Referrer = request.Site.Ref // Effectively the previous page to the page where the ad will be shown
-		body.Url = request.Site.Page     // where the impression will be made
-	}
-
-	gdpr := bidGdpr{}
-
-	ccpaPolicy, err := ccpa.ReadFromRequest(request)
+	bodyBytes, err := json.Marshal(request)
 	if err != nil {
-		errs = append(errs, err)
-	} else {
-		body.CCPA = ccpaPolicy.Consent
-	}
-
-	if request.Regs != nil && request.Regs.Ext != nil {
-		var extRegs openrtb_ext.ExtRegs
-		if err := json.Unmarshal(request.Regs.Ext, &extRegs); err != nil {
-			errs = append(errs, err)
-		} else {
-			if extRegs.GDPR != nil {
-				applies := *extRegs.GDPR != 0
-				gdpr.Applies = &applies
-				body.GDPR = &gdpr
-			}
-		}
+		return nil, []error{err}
 	}
 
-	if request.User != nil && request.User.Ext != nil {
-		var extUser openrtb_ext.ExtUser
-		if err := json.Unmarshal(request.User.Ext, &extUser); err != nil {
-			errs = append(errs, err)
-		} else {
-			gdpr.Consent = extUser.Consent
-			body.GDPR = &gdpr
-
-			if hasEids(extUser.Eids) {
-				body.User.Eids = extUser.Eids
-			}
+	if request.Site != nil {
+		requests := []*adapters.RequestData{
+			{
+				Method:  "POST",
+				Uri:     "https://e.serverbid.com/sb/rtb",
+				Body:    bodyBytes,
+				Headers: headers,
+			},
 		}
-	}
-
-	if request.Source != nil && request.Source.Ext != nil {
-		var extSChain openrtb_ext.ExtRequestPrebidSChain
-		if err := json.Unmarshal(request.Source.Ext, &extSChain); err != nil {
-			errs = append(errs, err)
-		} else {
-			body.SChain = extSChain.SChain
-		}
-	}
-
-	body.Coppa = request.Regs != nil && request.Regs.COPPA > 0
-
-	if request.Regs != nil && request.Regs.GPP != "" {
-		body.GPP = request.Regs.GPP
-	}
-
-	if request.Regs != nil && request.Regs.GPPSID != nil {
-		body.GPPSID = request.Regs.GPPSID
-	}
-
-	if request.Site != nil && request.Site.Content != nil {
-		body.Content = request.Site.Content
-	} else if request.App != nil && request.App.Content != nil {
-		body.Content = request.App.Content
-	}
-
-	for i, impression := range request.Imp {
-
-		_, consumableExt, err := extractExtensions(impression)
+		return requests, errs
 
+	} else {
+		_, consumableExt, err := extractExtensions(request.Imp[0])
 		if err != nil {
 			return nil, err
 		}
-
-		// These get set on the first one in observed working requests
-		if i == 0 {
-			body.NetworkId = consumableExt.NetworkId
-			body.SiteId = consumableExt.SiteId
-			body.UnitId = consumableExt.UnitId
-			body.UnitName = consumableExt.UnitName
-		}
-
-		body.Placements[i] = placement{
-			DivName:   impression.ID,
-			NetworkId: consumableExt.NetworkId,
-			SiteId:    consumableExt.SiteId,
-			UnitId:    consumableExt.UnitId,
-			UnitName:  consumableExt.UnitName,
-			AdTypes:   getSizeCodes(impression.Banner.Format), // was adTypes: bid.adTypes || getSize(bid.sizes) in prebid.js
+		var placementId = consumableExt.PlacementId
+		requests := []*adapters.RequestData{
```

Comment:
static/bidder-params/consumable.json mentions `placementId` is one of required param
```
 "oneOf": [
    {"required": ["siteId", "networkId","unitId"] },
    {"required": ["placementId"] }
  ]
```

based on adapter code  `placementId` is needed only for non-site media type request. 

`oneOf`  conditon will allow non-site media type request without  `placementId`
should add check here to make sure `PlacementId` is available  non-site media type request.